src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift

```diff
@@ -44,24 +44,38 @@ class ExposureSubmissionCoordinator: NSObject, RequiresAppDependencies {
 
 	func start(with coronaTestType: CoronaTestType? = nil) {
 		model.coronaTestType = coronaTestType
+		start(with: getInitialViewController())
+	}
 
-		let initialVC = getInitialViewController()
-		guard let parentNavigationController = parentNavigationController else {
-			Log.error("Parent navigation controller not set.", log: .ui)
-			return
-		}
+	func start(with testInformationResult: Result<CoronaTestQRCodeInformation, QRCodeError>) {
+		model.exposureSubmissionService.loadSupportedCountries(
+			isLoading: { _ in },
+			onSuccess: { supportedCountries in
+				switch testInformationResult {
+				case let .success(testInformation):
+					let qrInfoScreen = self.makeQRInfoScreen(supportedCountries: supportedCountries, testInformation: testInformation)
+					self.start(with: qrInfoScreen)
+				case let .failure(qrCodeError):
+					if qrCodeError == .invalidTestCode {
+						self.showRATInvalidQrCodeAlert()
+					}
+				}
+			}
+		)
+	}
 
-		/// The navigation controller keeps a strong reference to the coordinator. The coordinator only reaches reference count 0
-		/// when UIKit dismisses the navigationController.
-		let exposureSubmissionNavigationController = ExposureSubmissionNavigationController(
-			coordinator: self,
-			dismissClosure: { [weak self] in
-				self?.dismiss()
-			},
-			rootViewController: initialVC
+	private func showRATInvalidQrCodeAlert() {
```

Comment:
Maybe a bit niggling, but in the localized strings, we do not camelcase `QR`, in the function name `showRATInvalidQrCodeAlert` we mix both. So might `showRATInvalidQRCode` be an option? There are function name examples in our codebase (eg. `.latestQRCodePosterTemplate()`) where we also do not camelCase it. 
Even later down the code road in this file, there is a function `makeQRInfoScreen`.
Just thinking.